bazel/repositories.bzl

```diff
@@ -511,6 +517,58 @@ def _com_google_protobuf():
         actual = "@com_google_protobuf//util/python:python_headers",
     )
 
+def _io_opencensus_cpp():
+    location = REPOSITORY_LOCATIONS["io_opencensus_cpp"]
+    http_archive(
+        name = "io_opencensus_cpp",
+        patch_args = ["-p0"],
+        patches = ["@envoy//bazel/foreign_cc:io_opencensus_cpp.patch"],
+        **location
+    )
+    native.bind(
+        name = "opencensus_trace",
+        actual = "@io_opencensus_cpp//opencensus/trace",
+    )
+    native.bind(
+        name = "opencensus_trace_cloud_trace_context",
+        actual = "@io_opencensus_cpp//opencensus/trace:cloud_trace_context",
+    )
+    native.bind(
+        name = "opencensus_trace_grpc_trace_bin",
+        actual = "@io_opencensus_cpp//opencensus/trace:grpc_trace_bin",
+    )
+    native.bind(
+        name = "opencensus_trace_trace_context",
+        actual = "@io_opencensus_cpp//opencensus/trace:trace_context",
+    )
+    native.bind(
+        name = "opencensus_exporter_stdout",
+        actual = "@io_opencensus_cpp//opencensus/exporters/trace/stdout:stdout_exporter",
+    )
+    native.bind(
+        name = "opencensus_exporter_stackdriver",
+        actual = "@io_opencensus_cpp//opencensus/exporters/trace/stackdriver:stackdriver_exporter",
+    )
+    native.bind(
+        name = "opencensus_exporter_zipkin",
+        actual = "@io_opencensus_cpp//opencensus/exporters/trace/zipkin:zipkin_exporter",
+    )
+
+def _com_github_curl():
+    # Used by OpenCensus Zipkin exporter.
+    location = REPOSITORY_LOCATIONS["com_github_curl"]
+    http_archive(
+        name = "com_github_curl",
+        build_file_content = BUILD_ALL_CONTENT + """
+cc_library(name = "curl", visibility = ["//visibility:public"], deps = ["@envoy//bazel/foreign_cc:curl"])
```

Comment:
Why is the additional `cc_libtrary` beyond what is provided by `foreign_cc` needed here? Isn't the `bind` sufficient?

Reply:
OpenCensus depends on `@com_gihub_curl//:curl` instead of `//external:curl`(?)

I chose to keep the former. If you prefer, I could add more patches to io_opencensus_cpp.patch to use the latter.
 
Without build_file_content, there is no curl package and I get:
```
ERROR: external/io_opencensus_cpp/opencensus/exporters/trace/zipkin/BUILD:24:1: no such package '@com_github_curl//': BUILD file not found on package path and referenced by '@io_opencensus_cpp//opencensus/exporters/trace/zipkin:zipkin_exporter'
```

What do you think?

Reply:
It's probably fine as is, but it would be good to have a plan to undo this hackiness in the future. Some people keep saying that `bind` is going away, but I haven't seen any evidence of this actually happening, and TBH it's probably going to be around for a long time until we have completely standard shared ideas of paths for every repo in the world globally (CC @jmillikin-stripe).

Reply:
I agree. Long-term I would like to see everyone use the same repo names and paths. In the short term, I'm not sure what to do in order to make the situation better instead of worse.

[example](https://github.com/census-instrumentation/opencensus-cpp/tree/master/tools/zlib)